### DISCUSSION TITLE
Welcome to azkar-remider-chrome-extension Discussions!

### DISCUSSION THREAD

**Discussion by nagwan** [nagwan/azkar-remider-chrome-extension#9] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by m-1226** [comment]
السلام عليكم ورحمه الله وبركاته 
فيه بعض الاقتراحات علي الكود هي حاجات بسيطة
بالنسبة للوقت في الoptions 
يفضل يكون ( ١ ، ٣ ، ٥ ، ٧ ) دقيقة وممكن توصل لحد ٢٠ دقيقة 
مش الوقت المعروض واللي هو أقل حاجة تقريبا ١٥ دقيقة 
يعني الذكر كل ما كان وقته أقل ودائم يكون افضل حتي للي هيقول الذكر 
لكن المدة الكبيرة واللي تقريبا بتوصل لساعة في الoptions يعني من وجهة نظري المتواضعة مش افضل شي 
حتي تطبيقات الاذكار كلها اللي بستخدمها
بيكون الوقت مابين ١ - ١٥ دقيقة علشان دوام ذكر الله وعامل الوقت في حاجة زي كده اهم شي.

**Comment by m-1226** [comment]
الاقتراح الثاني 
انا بدات اعدل علي الكود بالفعل واول ما اخلص اول جزء هعمل pull باذن الله 

 النقطة بس  اللي فعلا حابب أنها تنفذ 

إن الأداة تكون باللغة العربية 
ده افضل كتيير جدا جدا 
لان الذكر بالعربي وكده كده الأداة بتستهدف اللي بيعرف اللغة العربية 
ف اني اخلي كل حاجة علي نسق واتجاه واحد اكيد افضل 
ده رايي المتواضع

**Comment by m-1226** [comment]
السلام عليكم ورحمة الله وبركاته 
الحمدلله عملت بعض التعديلات البسيطة علي الإضافة وأرجو إنها تكون كويسة ولو بشق تمرة 
1- عدلت ملف azkar.js بالكامل بإني ضفت ثلاثة مفاتيح علشان اعرض الذكر وهم ( tag, text , belss ) ال tag بيكون مكتوب فيه هو الذكر أذكار صباح ولا مساء وده بستخدمه في الbackground.js لإني عامل function علي اساس الوقت الحالي يعرض أذكار الصباح ولا المساء بناء علي الtag 
والtext فيه الذكر 
اما الbless فهي وصف الذكر او بركة الذكر او الفائدة اللي هتعود عليك منه كما ورد عن الرسول صلي الله عليه وسلم ( مثلا من قالها لم يضره شئ ) 
2- خليت الnotification بدل ما تظهر وتختفي سريعا إنها تتطلب الuser interaction او تفاعل منه بإنه يضغط close وبالتالي ده يكون أكثر حرصا علي قراءة الذكر 
3- ضفت إشعار بسيط كده مش بيتشغل غير كل يوم جمعة يذكره بقراءة سورة الكهف وفضل قراءتها في اليوم ده فقط يعني مش بتظهر في اي يوم تاني 
4- خليت الذكر يتعرض اه بشكل عشوائي كما كان ولكن نفس الذكر ميتكررش مرتين لحد لما يخلص ملف الأذكار ويعيد الكرة مرة تانية لإنه ساعات الإضافة بتعمل الذكر بشكل عشوائي مظبوط ولكنها بتفضل علي ذكر معين لمدة كده ومش بتجيب غيره ولكن بعد التعديل اللي تم كده هي بإذن الله هتجيب الذكر بشكل عشوائي ولكن هيكون شرط unique إنه ميكنش إتعرض سابقا وده تم بإني ضفت list إسمها last_zekr بتحفظ الzekr اللي طلع وبتعمل check هل هو نفس اللي هيتطلع ولا لا , إذا طلع متشابه تعرض غيره وإذا طلع لا مش متشابه ف تعرضه عادي 
5- عدلت إسم الإضافة ل  وَذْكَر  ولقيت إنه افضل من وجهة نظري المتواضعة لإنه أولا إسلامي وعربي وفي نفس الوقت بيعبر عن اللي بيقدمه  

الإضافات دي كلها مينفعش اعملها pull request إلا لما فقط يتم الموافقة علي الpull request  اللي قبله علشان بس يعني التنظيم و معرفة الخطوات أول بأول 

بس كده 
بعتذر علي طول الرسالة والإزعاج 
والسلام عليكم ورحمة الله وبركاته

**Comment by m-1226** [comment]
السلام عليكم ورحمة الله وبركاته
الحمدلله أعدت تصميم صفحة الإعدادات بحيث يقدر يغير وقت ظهور الأذكار ( 15 , 20 , 30 , 45 , 60 )
و ده عرض للصفحة

![image](https://github.com/nagwan/azkar-remider-chrome-extension/assets/64734981/a497fdd7-4e54-4ece-a9b7-6f6ce159f8e7)

بيختار منها الوقت وبعدها بيظهر رسالة إنه تم حفظ الإعدادات.

خليت كمان الوصول إلي صفحة الإعدادات أسهل بإني ضفت أيقونة علي الشمال فوق في الإضافة ( شكل ترس ) يضغط عليها تأخذه مباشرة لصفحة الإعدادات بحيث يكون أسهل عليه إذا مكانش عارف يوصل ليها من فين.
بس كده
بعتذر علي إزعاجي
